pkg/tcpip/stack/stack.go

```diff
@@ -46,6 +48,41 @@ const (
 	resolutionAttempts = 3
 )
 
+const (
+	// fakeNetNumber is used as a protocol number in tests.
+	// This constant should match fakeNetNumber in stack_test.go
+	fakeNetNumber    tcpip.NetworkProtocolNumber = math.MaxUint32
+)
+
+// Packet forwarding flags. Forwarding settings for different network protocols
+// are stored as bit flags in an uint32 number.
+const (
+	forwardingIPv4 uint32 = 1 << iota
+	forwardingIPv6
+
+	// forwardingFake is used to test package forwarding with a fake protocol.
+	forwardingFake
+)
+
+func getForwardingFlag(protocol tcpip.NetworkProtocolNumber) uint32 {
+	var flag uint32
+	switch protocol {
+	case header.IPv4ProtocolNumber:
+		flag = forwardingIPv4
+
+	case header.IPv6ProtocolNumber:
+		// IP forwarding in master branch is always off. There is no way to change ipv6 forwarding.
+		flag = forwardingIPv6
+
+	case fakeNetNumber:
+		// This network protocol number is used in stack_test to test packet forwarding.
+		flag = forwardingFake
+	default:
+		// Disable packet forwarding for others protocols.
```

Comment:
I am not quire sure that I follow. Is this a TODO?

Reply:
yep, It's a good question) As I understand now the better way to deal with this TODO was to open discussion first. 
In the master branch, the netstack contains three network protocols: arp, ipv4, ipv6.  Packet forwarding is enabled/disabled by a single bool flag. And original TODO is to expose the flag via /proc/sys/net/ipv4/ip_forward.

This PR 1) removes the single flag, and packet forwarding can be enabled/disabled independently for every protocol: ipv4/ipv6 2) fixes a problem with not enough space to send a packet to a link-layer. 

And I think that there are a couple of issues with the code in nic.go behind "if n.stack.Forwarding(protocol)":
1) it should decrease ttl 
2) it should handle multicasts

Should I address the issues in this PR? 
btw, I am not an expert in network programming, and my development speed is very limited; if my PR doesn't help you and only spends your time plz feel free to close it.

Reply:
No need to try to fix the limitations of the existing implementation. What we normally do in cases like this is file an issue and add a TODO to the code referencing the issue. This opens a discussion about the issue where possible solutions can be discussed (and if the issue is actually a problem).

This PR is looking very good and is almost ready to be merged. Thanks for the great work!

Reply:
I've put a couple of TODOs in NIC.DeliverNetworkPacket

Reply:
I see one new TODO in (*NIC).DeliverNetworkPacket (looks good). Add the part about TTLs?

Also, there is still the issue here about other protocols. The meaning of this comment is somewhat unclear to me. Maybe something like this would be better?
`We only support forwarding for IPv4 and IPv6.`

Reply:
fixed